pkg/apis/eventing/v1alpha1/trigger_lifecycle.go

```diff
@@ -80,3 +83,51 @@ func (ts *TriggerStatus) PropagateSubscriptionStatus(ss *SubscriptionStatus) {
 func (ts *TriggerStatus) MarkNotSubscribed(reason, messageFormat string, messageA ...interface{}) {
 	triggerCondSet.Manage(ts).MarkFalse(TriggerConditionSubscribed, reason, messageFormat, messageA...)
 }
+
+func (ts *TriggerStatus) SetAddress(url *apis.URL) {
+	if url != nil {
+		ts.Address.URL = url
+		brokerCondSet.Manage(ts).MarkTrue(BrokerConditionAddressable)
+	} else {
+		ts.Address.URL = nil
+		brokerCondSet.Manage(ts).MarkFalse(BrokerConditionAddressable, "emptyHostname", "hostname is the empty string")
+	}
+}
+
+func (ts *TriggerStatus) extendImporters(index int) {
+	if index >= len(ts.Importers) {
+		ts.Importers = append(ts.Importers, make([]TriggerImporterStatus, index-len(ts.Importers)+1)...)
+	}
+}
+
+func (ts *TriggerStatus) MarkImporter(index int, importer v1.ObjectReference) {
+	ts.extendImporters(index)
+	ts.Importers[index].Ref = &importer
+}
+
+func (ts *TriggerStatus) MarkImportersFailed(err error) {
+	// something
+}
+
+func (ts *TriggerStatus) MarkImporterRef(index int, importer corev1.ObjectReference) {
+	ts.extendImporters(index)
+	ts.Importers[index].Ref = &importer
+}
+
+func (ts *TriggerStatus) PropagateImporterStatus(index int, status runtime.RawExtension) {
+	ts.extendImporters(index)
+	ts.Importers[index].Status = &status
```

Comment:
I'm seeing the Trigger's `sinkUri` duplicated in each importer status. It might be a better experience for the user if the Trigger status only propagates the Importer's conditions array rather than the entire status.

Looking elsewhere, this will be a lot of tedious mucking around with unstructured.Unstructured...